### PR TITLE
NWB updates from `run_kilosort` branch to main

### PR DESCRIPTION
**Pull request by CBroz1**
Updates 
- enforce `ephys_mode` via dj.config rather than activated schema, which had complications in testing.
- limit comments to 88 char
- fix bug on creating units table

I chose not to version bump as this would make the main branch's changelog out of synch with the dev branch

### CODE REVIEW

File: element_array_ephys/export/nwb/nwb.py [hunk 1]
```diff
@@ -356,18 +361,17 @@ def add_ephys_recording_to_nwb(
...
 
     source data -> acquisition["ElectricalSeries"]
 
     Parameters
     ----------
-    session_key: dict
-    ephys_root_data_dir: str
-    nwbfile: NWBFile
-    end_frame: int, optional
+    session_key: dict ephys_root_data_dir: str nwbfile: NWBFile end_frame: int, optional
```
Comment (kabilar) [inline element_array_ephys/export/nwb/nwb.py:374]:
```suggestion
    session_key: dict
    ephys_root_data_dir: str
    nwbfile: NWBFile
    end_frame: int, optional
```

File: element_array_ephys/export/nwb/nwb.py [hunk 2]
```diff
@@ -14,12 +14,14 @@
...
 import warnings
-from ... import probe, ephys_no_curation
+from ... import probe
+from ... import ephys_no_curation as ephys
 
-assert probe.schema.is_activated(), 'probe not yet activated'
 
-assert ephys_no_curation.schema.is_activated,  \
-        "The ephys module must be activated before export."
+ephys_mode = os.getenv('EPHYS_MODE', dj.config['custom'].get('ephys_mode', 'acute'))
```
Comment (kabilar) [inline element_array_ephys/export/nwb/nwb.py:21]:
```suggestion
ephys_mode = os.getenv('EPHYS_MODE', dj.config['custom'].get('ephys_mode'))
```

Comment (kabilar) [reply element_array_ephys/export/nwb/nwb.py:21]: If no value is specified, should we set the default to `None` and then the `NotImplementedError` will also be thrown?

File: element_array_ephys/export/nwb/nwb.py [hunk 3]
```diff
@@ -215,7 +219,8 @@ def create_units_table(
             ephys.ProbeInsertion
             * ephys.CuratedClustering.Unit
             * probe.ProbeType.Electrode
-            & unit
+            & dict((k, unit[k]) for k in unit.keys()  # excess keys caused errs
+                    if k not in ['spike_times', 'spike_sites', 'spike_depths'])
```
Comment (kabilar) [inline element_array_ephys/export/nwb/nwb.py:223]:
```suggestion
            & dict((k, unit[k]) for k in unit.keys()  # excess keys caused errs
                    if k not in ['cluster_quality_label', 'spike_count', 'spike_times', 'spike_sites', 'spike_depths'])
```

Comment (kabilar) [reply element_array_ephys/export/nwb/nwb.py:223]: Or do we need to restrict by the primary key of `ephys.CuratedClustering.Unit`?  Please carefully check this suggestion.  I may be misinterpreting something.

**Review by kabilar — CHANGES_REQUESTED** [review]
Thanks @CBroz1.  Great work!

File: element_array_ephys/export/nwb/nwb.py (hunk 3 above)
Comment (CBroz1) [reply element_array_ephys/export/nwb/nwb.py:223]: My recollection is that I excluded exactly the keys that were needed, but I can retest with this suggestion

File: element_array_ephys/export/nwb/nwb.py (hunk 2 above)
Comment (CBroz1) [reply element_array_ephys/export/nwb/nwb.py:21]: This mirrors `acute` as default in the workflow, but either way works in the present case

File: element_array_ephys/export/nwb/nwb.py [hunk 4]
```diff
@@ -215,7 +219,8 @@ def create_units_table(
             ephys.ProbeInsertion
             * ephys.CuratedClustering.Unit
             * probe.ProbeType.Electrode
-            & unit
+            & dict((k, unit[k]) for k in unit.keys()  # excess keys caused errs
```
Comment (dimitri-yatsenko) [inline element_array_ephys/export/nwb/nwb.py:222]:
```suggestion
            & {k: v for k, v in unit.items()  # remove extra keys
```

**Comment by ttngu207** [comment]
Fully incorporated in #94

**Pull request closed without merging**